frontend/packages/component-library/src/cms/heroBanner/HeroBanner.tsx

```diff
@@ -0,0 +1,111 @@
+import classNames from 'classnames';
+import {ReactNode, HTMLAttributes} from 'react';
+
+import {LinkButton, LinkButtonProps} from '@/button';
+import {Theme} from '@/common/contexts';
+import Image, {ImageProps} from '@/image';
+import {Heading1, BodyOneText, BodyTwoText} from '@/typography';
+import Video, {VideoProps} from '@/video';
+
+import moduleStyles from './heroBanner.module.scss';
+
+export interface HeroBannerProps extends HTMLAttributes<HTMLElement> {
+  /** HeroBanner heading */
+  heading: string | ReactNode;
+  /** HeroBanner subHeading */
+  subHeading?: string | ReactNode;
+  /** HeroBanner description */
+  description?: string | ReactNode;
+  /** HeroBanner image */
+  imageProps?: ImageProps;
+  /** HeroBanner video */
+  videoProps?: VideoProps;
+  /** HeroBanner link */
+  buttonProps?: LinkButtonProps;
+  /** HeroBanner custom background url */
+  backgroundImageUrl?: string;
+  /** HeroBanner theme value.
+   *  If you're using backgroundImageUrl - you should make sure you set correct theme value to HeroBanner.
+   *  */
+  'data-theme'?: Theme;
+  /** Whether to show the background color */
+  removeBackground?: boolean;
+  /** HeroBanner whether show with wide text (text is wider than image) */
+  withWideText?: boolean;
+  /** HeroBanner partner prop */
+  partner?: {title: string; logo: ImageProps};
+  /** HeroBanner custom className  */
+  className?: string;
+}
+
+/**
+ * ## Production-ready Checklist:
+ *  * (✔) implementation of component approved by design team;
+ *  * (✔) has storybook, covered with stories and documentation;
+ *  * (✔) has tests: test every prop, every state and every interaction that's js related;
+ *  * (see ./__tests__/HeroBanner.test.tsx)
+ *  * (✔) passes accessibility checks;
+ *
+ * ### Status: ```Ready for dev```
+ *
+ * Design System: HeroBanner Component.
+ * Renders a Hero Banner/Section which serves as an opening section of a page.
+ */
+const HeroBanner: React.FC<HeroBannerProps> = ({
+  heading,
+  subHeading,
+  description,
+  partner,
+  buttonProps,
+  imageProps,
+  videoProps,
+  backgroundImageUrl,
+  removeBackground = false,
+  withWideText = false,
+  className,
+  ...HTMLAttributes
+}) => (
+  <section
+    {...HTMLAttributes}
+    className={classNames(moduleStyles.heroBannerWrapper, className, {
+      [moduleStyles['heroBanner-withWideText']]: withWideText,
+    })}
+    data-theme={HTMLAttributes['data-theme']}
+    style={{
+      ...(HTMLAttributes.style ?? {}),
+      backgroundImage: backgroundImageUrl
+        ? `url(${backgroundImageUrl})`
+        : HTMLAttributes.style?.backgroundImage,
+      ...(removeBackground ? {background: 'none'} : {}),
+    }}
+  >
+    <div className={classNames(moduleStyles.heroBannerContainer)}>
+      <div className={moduleStyles.heroBannerDescriptiveSection}>
```

Comment:
Super nit, feel free to ignore! I usually call this the `textWrapper` on these banners since it's more concise. 
```suggestion
      <div className={moduleStyles.heroBannerTextWrapper}>
```

Reply:
I'm not usre if Section is ideal name here  , but I thought Container is too much since these are rather smaller parts 😄
But if choosing between wrapper and container I think it might be better to call this `Text/Descriptive Container`
What do you think?

Reason for this:
https://css-tricks.com/best-way-implement-wrapper-css/
`In programming languages, the word container is generally used for structures that can contain more than one element. A wrapper, on the other hand, is something that wraps around a single object to provide more functionality and interface to it.`

Reply:
Oof this is tricky haha! I think In this case `wrapper` works since the text content group could be considered a single object (like you wouldn't wrap just the `h1` in a wrapper), and container works for the broader container above it. For consistency I lean towards `TextWrapper`.

Reply:
> I think In this case wrapper works since the text content group could be considered a single object (like you wouldn't wrap just the h1 in a wrapper)

It's rather a question of structure then semantics. (Here each container has multiple elements inside of it)
e.g. h1 with wrapper might be not ideal example, but still h1 could have a wrapper theoretically.


example:

```
<div clasName='wrapper1'>
  <div className='container'>
    <div className='wrapper2'>
      <h1 className='heading'>some text</h1>
    </div>
    <p>someText</p>
    <span>someText</span>
  </div>
</div>
```

I will keep it textContainer and mediaContainer for now and can update it later if needed. But I'm definitely onboard with making things consistent so what do you think if we discuss it on check in or in slack team thread maybe? So that we add one more rule to our naming conventions

Reply:
I think it's worth a quick discussion in check-in 🙌